e2e_tests/test_workspace_services.py

```diff
@@ -63,76 +44,21 @@ async def test_create_guacamole_service_into_base_workspace(verify) -> None:
         }
     }
 
-    user_resource_path, user_resource_id = await post_resource(user_resource_payload, f'/api{workspace_service_path}/{strings.API_USER_RESOURCES}', workspace_owner_token, verify, method="POST")
+    user_resource_path, _ = await post_resource(user_resource_payload, f'/api{workspace_service_path}/{strings.API_USER_RESOURCES}', workspace_owner_token, verify, method="POST")
 
     await disable_and_delete_resource(f'/api{user_resource_path}', workspace_owner_token, verify)
 
     await disable_and_delete_resource(f'/api{workspace_service_path}', workspace_owner_token, verify)
 
-    admin_token = await get_admin_token(verify)
-    await disable_and_delete_resource(f'/api{workspace_path}', admin_token, verify)
-
 
 @pytest.mark.extended_aad
 @pytest.mark.timeout(75 * 60)
-async def test_create_guacamole_service_into_aad_workspace(verify) -> None:
+async def test_create_guacamole_service_into_aad_workspace(verify, setup_test_aad_workspace) -> None:
     """This test will create a Guacamole service but will create a workspace and automatically register the AAD Application"""
+    workspace_path, workspace_id, workspace_owner_token = setup_test_aad_workspace
 
-    payload = {
-        "templateName": strings.BASE_WORKSPACE,
-        "properties": {
-            "display_name": "E2E test guacamole service",
-            "description": "workspace for E2E AAD",
-            "address_space_size": "small",
-            "auth_type": "Automatic"
-        }
-    }
-    if config.TEST_WORKSPACE_APP_PLAN != "":
-        payload["properties"]["app_service_plan_sku"] = config.TEST_WORKSPACE_APP_PLAN
-
-    admin_token = await get_admin_token(verify)
-    workspace_path, workspace_id = await post_resource(payload, strings.API_WORKSPACES, access_token=admin_token, verify=verify)
-    workspace_owner_token, scope_uri = await get_workspace_auth_details(admin_token=admin_token, workspace_id=workspace_id, verify=verify)
-
-    service_payload = {
-        "templateName": strings.GUACAMOLE_SERVICE,
-        "properties": {
-            "display_name": "Workspace service test",
-            "description": "Workspace service for E2E test"
-        }
-    }
-
-    workspace_service_path, workspace_service_id = await post_resource(service_payload, f'/api{workspace_path}/{strings.API_WORKSPACE_SERVICES}', workspace_owner_token, verify)
-
-    await ping_guacamole_workspace_service(workspace_id, workspace_service_id, verify)
-
-    # patch the guac service. we'll just update the display_name but this will still force a full deployment run
-    # and essentially terraform no-op
-    patch_payload = {
-        "properties": {
-            "display_name": "Updated Guac Name",
-        }
-    }
-
-    await post_resource(patch_payload, f'/api{workspace_service_path}', workspace_owner_token, verify, method="PATCH")
```

Comment:
I don't think doing this is necessary for this test, as if AAD creation doesn't work, it should fail at the step of workspace creation

Reply:
Are you referring to the reason why it should be enough to just test the workspace creation step without any services etc?

Reply:
Yes

Reply:
I think it does - auto_create impacts the workspace_app settings and that is used by guacamole. 
If something doesn't work right there, the ping should fail (in theory at least). We can skip on VM but I think deploying/pinging Guacamole is still valuable.

Reply:
Ah okay, good to know. I'll add Guacamole back, and add a ping for Guacamole